gcc/rust/lint/rust-lint-scan-deadcode.h

```diff
@@ -79,6 +79,18 @@ class ScanDeadcode : public MarkLiveBase
       }
   }
 
+  void visit (HIR::TupleStruct &stct) override
+  {
+    // only warn tuple struct unconstructed, and ignoring unused field
+    HirId hirId = stct.get_mappings ().get_hirid ();
+    if (should_warn (hirId))
+      {
+	rust_warning_at (stct.get_locus (), 0, "%s is never %s: %<%s%>",
+			 "struct", "constructed",
```

Comment:
I think doing this makes it harder for people to search for an error's source in the code. Why not do 
```suggestion
	rust_warning_at (stct.get_locus (), 0, "struct is never constructed: %<%s%>",
```
?

Reply:
good point. i'll open another cleanup issue since there are some other warning using this way in my pass.